engine/collection/synchronization/engine.go

```diff
@@ -372,72 +334,64 @@ CheckLoop:
 		select {
 		case <-e.unit.Quit():
 			break CheckLoop
-		case <-poll.C:
-			err := e.pollHeight()
-			if err != nil {
-				e.log.Error().Err(err).Msg("could not poll heights")
-			}
-
+		case <-pollChan:
+			e.pollHeight()
 		case <-scan.C:
 			final, err := e.state.Final().Head()
 			if err != nil {
-				e.log.Error().Err(err).Msg("could not get final height")
+				e.log.Fatal().Err(err).Msg("could not get last finalized header")
 				continue
 			}
-
-			e.unit.Lock()
 			ranges, batches := e.core.ScanPending(final)
-			err = e.sendRequests(ranges, batches)
-			if err != nil {
-				e.log.Error().Err(err).Msg("could not send requests")
-			}
-			e.unit.Unlock()
+			e.sendRequests(ranges, batches)
 		}
 	}
 
 	// some minor cleanup
 	scan.Stop()
-	poll.Stop()
 }
 
 // pollHeight will send a synchronization request to three random nodes.
-func (e *Engine) pollHeight() error {
-
-	// get the last finalized header
-	final, err := e.state.Final().Head()
+func (e *Engine) pollHeight() {
+	head, err := e.state.Final().Head()
```

Comment:
Could we instead use `FinalizedHeaderCache`, like what is done in the common sync engine?

Reply:
Not really, collection cluster don't support same notification infrastructure as consensus nodes. Don't think it's worth implementing it only for this.